e2e-test/log/src/integrationTest/java/org/opensearch/dataprepper/integration/log/EndToEndOtelLogsSourceTest.java

```diff
@@ -0,0 +1,143 @@
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.dataprepper.plugins.sink.opensearch.ConnectionConfiguration;
+import org.opensearch.dataprepper.plugins.source.loggenerator.ApacheLogFaker;
+import org.opensearch.search.SearchHits;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.SessionProtocol;
+
+import io.netty.util.AsciiString;
+import io.opentelemetry.proto.collector.logs.v1.ExportLogsServiceRequest;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.logs.v1.LogRecord;
+import io.opentelemetry.proto.logs.v1.ResourceLogs;
+import io.opentelemetry.proto.logs.v1.ScopeLogs;
+
+public class EndToEndOtelLogsSourceTest {
+    private static final int SOURCE_PORT = 2021;
+    private static final String INDEX_NAME = "otel-logs-index";
+
+    private final ApacheLogFaker apacheLogFaker = new ApacheLogFaker();
+
+    @Test
+    public void testOtelLogsSourcePipelineEndToEnd() throws InvalidProtocolBufferException {
+        sendHttpRequestToSourceAndAssertResponse(SOURCE_PORT, createOtelLogsRequest());
+        final RestHighLevelClient restHighLevelClient = prepareOpenSearchRestHighLevelClient();
+        final List<Map<String, Object>> retrievedDocs = new ArrayList<>();
+
+        makeRequestAndAssertResponse(restHighLevelClient, retrievedDocs);
+    }
+
+    private HttpData createOtelLogsRequest() throws InvalidProtocolBufferException {
+        ExportLogsServiceRequest exportLogsServiceRequest = ExportLogsServiceRequest.newBuilder().addResourceLogs(
+                ResourceLogs.newBuilder()
+                        .addScopeLogs(ScopeLogs.newBuilder()
+                                .addLogRecords(LogRecord.newBuilder()
+                                        .setBody(AnyValue.newBuilder().setStringValue(apacheLogFaker.generateRandomCommonApacheLog()).build())
+                                        .setSeverityNumberValue(1)
+                                        .setTimeUnixNano(System.currentTimeMillis() * 1_000_000)
+                                        .setTraceId(ByteString.copyFromUtf8("trace-id"))
+                                        .setSpanId(ByteString.copyFromUtf8("span-id")
+                                )).build()
+                        )
+        ).build();
+
+        return HttpData.copyOf(JsonFormat.printer().print(exportLogsServiceRequest).getBytes());
```

Comment:
This creates just the JSON representation. I think we also need the protobuf version.